### ISSUE TITLE
Update with --prefer-lowest errors but shouldn't

### ISSUE THREAD

**Issue by alexander-schranz** [composer/composer#7161]
My `composer.json` based on the requirements from [doctrine/common](https://github.com/doctrine/common/blob/v2.5.0/composer.json#L20) and [swiftmailer/swiftmailer](https://github.com/swiftmailer/swiftmailer/blob/v6.0.0/composer.json#L19)

```json
{
    "require": {
        "doctrine/lexer": "1.*",
        "egulias/email-validator": "~2.0"
    }
}
```

Output of `composer diagnose`:

```
Checking composer.json: OK
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com oauth access: OK
Checking disk free space: OK
Checking pubkeys:
Tags Public Key Fingerprint: -
Dev Public Key Fingerprint: -
OK
Checking composer version: OK
Composer version: 1.6.3 (last tested with 1.8.0 (error still exists))
PHP version: 7.1.14
PHP binary path: /usr/local/Cellar/php71/7.1.14_25/bin/php
```

When I run this command:

```
composer update --prefer-lowest
```

I get the following output:

```
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - egulias/email-validator 2.0.0 requires doctrine/lexer ^1.0.1 -> satisfiable by doctrine/lexer[v1.0.1].
    - egulias/email-validator 2.0.1 requires doctrine/lexer ^1.0.1 -> satisfiable by doctrine/lexer[v1.0.1].
    - egulias/email-validator 2.1.0 requires doctrine/lexer ^1.0.1 -> satisfiable by doctrine/lexer[v1.0.1].
    - egulias/email-validator 2.1.1 requires doctrine/lexer ^1.0.1 -> satisfiable by doctrine/lexer[v1.0.1].
    - egulias/email-validator 2.1.2 requires doctrine/lexer ^1.0.1 -> satisfiable by doctrine/lexer[v1.0.1].
    - egulias/email-validator 2.1.3 requires doctrine/lexer ^1.0.1 -> satisfiable by doctrine/lexer[v1.0.1].
    - Conclusion: don't install doctrine/lexer v1.0.1
    - Installation request for egulias/email-validator ~2.0 -> satisfiable by egulias/email-validator[2.0.0, 2.0.1, 2.1.0, 2.1.1, 2.1.2, 2.1.3].
```

And I expected this to happen:

```
  - Installing doctrine/lexer (v1.0.1): Loading from cache
  - Installing egulias/email-validator (2.0.0): Loading from cache
```

**Workaround by localheinz**

```bash
composer install
composer update --prefer-lowest
```

**Comment by localheinz** [comment]
@alexander-schranz 

I have similar issues!

Starting with

```
$ rm -rf vendor
```

I have been wondering whether

```
$ composer update --prefer-lowest
```

and

```
$ composer install && composer update --prefer-lowest
```

should achieve the same results.

I believe they should, but as observed a couple of times, they don't: I end up with similar errors, for example:

```
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - phpunit/phpunit 7.0.1 requires sebastian/diff ^3.0 -> satisfiable by sebastian/diff[3.0.0].
    - phpunit/phpunit 7.0.2 requires sebastian/diff ^3.0 -> satisfiable by sebastian/diff[3.0.0].
    - Conclusion: don't install sebastian/diff 3.0.0
    - Installation request for phpunit/phpunit ^7.0.1 -> satisfiable by phpunit/phpunit[7.0.1, 7.0.2].
```

**Comment by alexander-schranz** [comment]
@naderman confirmed does fix my issue:

![bildschirmfoto 2019-02-04 um 10 09 05](https://user-images.githubusercontent.com/1698337/52198963-ed06d780-2864-11e9-8e64-3452278cfd90.png)

**Comment by naderman** [comment]
Fixed on master / 1.8 now.

You can use a current composer snapshot with `composer self-update --snapshot`.
